frontend/test/pytest/test_measurements_results.py

```diff
@@ -160,6 +160,48 @@ def expval3(x: float):
         observed = expval3(np.pi)
         assert np.isclose(observed, expected)
 
+    def test_hermitian_shots_expval(self, backend):
+        """Test expval Hermitian observables with shots."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=3, shots=1000))
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            A = np.array(
+                [[complex(1.0, 0.0), complex(2.0, 0.0)], [complex(2.0, 0.0), complex(1.0, 0.0)]]
+            )
+            return qml.expval(qml.Hermitian(A, wires=2) + qml.PauliX(0) + qml.Hermitian(A, wires=1))
+
+        with pytest.raises(
+            RuntimeError,
+            match="For Hermitian Observables, expval calculation is not supported by shots",
+        ):
+            circuit(np.pi / 4, np.pi / 4)
+
+        qml.operation.disable_new_opmath()
+
+    def test_hermitian_shots_var(self, backend):
+        """Test var Hermitian observables with shots."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=3, shots=1000))
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            A = np.array(
+                [[complex(1.0, 0.0), complex(2.0, 0.0)], [complex(2.0, 0.0), complex(1.0, 0.0)]]
+            )
+            return qml.var(qml.Hermitian(A, wires=2) + qml.PauliX(0) + qml.Hermitian(A, wires=1))
+
+        with pytest.raises(
+            RuntimeError,
+            match="Hermitian observables do not support applyInPlaceShots method",
```

Comment:
@multiphaseCFD as discussed offline, there are some inconsistent error messages raising for 'lightning.qubit' and 'lightning.kokkos' for `expval`, `var`, and `probs`.

Reply:
Our current CI failures are related to this issue.

Reply:
How about setting the error message as `xxx observables do not have shots support.`?